src/python/pants/backend/python/util_rules/interpreter_constraints.py

```diff
@@ -67,13 +66,13 @@ def parse_constraint(constraint: str) -> Requirement:
     interpreter.py's `parse_requirement()`.
     """
     try:
-        parsed_requirement = Requirement.parse(constraint)
-    except ValueError as err:
+        parsed_requirement = Requirement(constraint)
+    except InvalidRequirement:
         try:
-            parsed_requirement = Requirement.parse(f"CPython{constraint}")
-        except ValueError:
+            parsed_requirement = Requirement(f"CPython{constraint}")
+        except InvalidRequirement as err2:
             raise InvalidRequirement(
-                f"Failed to parse Python interpreter constraint `{constraint}`: {err.args[0]}"
+                f"Failed to parse Python interpreter constraint `{constraint}`: {err2}"
```

Comment:
Not sure `err2` is what we want to show to the user. What we actually failed to parse (from a user perspective) is in the first error. Tacking `CPython` in front is an attempt to satisfy some shorthand, but if it fails too it'll be confusing to the user to show that string.

For example, suppose the constraint is `Foo$==3.4.5` because of a typo. `err2` will refer to `CPythonFoo$==3.4.5`, which is going to look bizarre.

Reply:
The background for this change was that it allowed me to better debug this migration when I had inadvertently left out the requirement name in the `and_constraints` function and the code failed when it tried to parse a requirement using a `<` operator with the `.*` pattern which is not permitted by standard. So one error obscured the other.

But that was just during the migration, I'll go back to the original logic as suggested.

Reply:
Done.